syft/pkg/cpe.go

```diff
@@ -99,6 +99,22 @@ func CPEString(c CPE) string {
 	return output.BindToFmtString()
 }
 
+func CPEURI(c CPE) string {
+	output := CPE{}
+	output.Vendor = sanitize(c.Vendor)
+	output.Product = sanitize(c.Product)
+	output.Language = sanitize(c.Language)
+	output.Version = sanitize(c.Version)
+	output.TargetSW = sanitize(c.TargetSW)
+	output.Part = sanitize(c.Part)
+	output.Edition = sanitize(c.Edition)
+	output.Other = sanitize(c.Other)
+	output.SWEdition = sanitize(c.SWEdition)
+	output.TargetHW = sanitize(c.TargetHW)
```

Comment:
This mimics our custom CPEString logic above to handle special characters in the CPE.